### PR TITLE
Bump coreos-assembler to Fedora 39

### PR DESCRIPTION
**Pull request by jlebon**
Matches https://github.com/coreos/coreos-assembler/pull/3653.

While we're here, use the Quay.io image instead since that's the future.

### CODE REVIEW

File: ci-operator/config/coreos/coreos-assembler/coreos-coreos-assembler-main.yaml [hunk 1]
```diff
@@ -1,7 +1,7 @@
 build_root:
   project_image:
     dockerfile_literal: |
-      FROM registry.fedoraproject.org/fedora:38
+      FROM registry.fedoraproject.org/fedora:39
```
Comment (dustymabe) [inline ci-operator/config/coreos/coreos-assembler/coreos-coreos-assembler-main.yaml:4]: should we update this to `quay.io/fedora/fedora:39` ?

**Comment by jlebon** [comment]
/pj-rehearse ack

We'll test this in https://github.com/coreos/coreos-assembler/pull/3653.

**Pull request merged**